nemo_skills/finetuning/filtering_utils.py

```diff
@@ -29,15 +29,25 @@ def process_bad_solutions(
     filtered_predictions = []
     code_solns = []
     text_solns = []
+    # Solution set to ensure uniqueness of solutions filtered through
+    solution_set = set()
     for sample in samples:
         if should_remove(sample['generation'], solution_filters):
             continue
         if should_trim:
             sample['generation'] = trim_output(sample['generation'])
+
+        if sample['generation'] in solution_set:
```

Comment:
So we actually do have deduplication initially, but it looks like you're getting some more duplicates after trimming. @i-vainn let's make sure that in your refactoring this deduplication is an explicit processor that we call last right before producing the final file to avoid this in the future.